database/plugin/metadata/sqlite/transaction.go

```diff
@@ -34,7 +35,13 @@ func (d *MetadataStoreSqlite) GetTransactionByHash(
 	if txn == nil {
 		txn = d.DB()
 	}
-	result := txn.First(ret, "hash = ?", hash)
+	result := txn.
+		Preload("Inputs").
+		Preload("Outputs").
+		Preload("ReferenceInputs").
+		Preload("Collateral").
+		Preload("CollateralReturn").
```

Comment:
Unless you're trying to limit the associated tables that are automatically preloaded, you could replace these 5 with a single call:

```
Preload(clause.Associations)
```

https://gorm.io/docs/preload.html#Preload-All